apps/vscode-extension/src/intellisense/configuration.ts

```diff
@@ -0,0 +1,64 @@
+import { CompletionItem, CompletionItemKind, CompletionItemProvider, SnippetString } from 'vscode';
+
+interface ConfigurationTags {
+  description: string;
+  detail: string;
+  snippet: string;
+}
+
+const configurationTags: Record<string, ConfigurationTags> = {
+  o3rWidget: {
+    description: 'Tag to use CMS widget for configuration',
+    detail: 'widgetName',
+    // TODO compute snippet thanks to eslint config to propose only valid widgetName
+    snippet: '${1:widgetName}'
+  },
+  o3rWidgetParam: {
+    description: 'Tag to use CMS widget parameter for configuration',
+    detail: 'paramName paramValue',
+    // TODO compute snippet thanks to eslint config to propose only valid paramName and paramValue
+    snippet: '${1:paramName} ${2:paramValue}'
+  }
+};
+
+const completions = Object.entries(configurationTags).map(([label, { detail, description, snippet }]) => {
+  const completion = new CompletionItem({ label: `${label} `, detail }, CompletionItemKind.Keyword);
+  completion.documentation = description;
+  completion.insertText = new SnippetString(`${label} ${snippet}`);
+
+  return completion;
+});
+
+export const configurationCompletionItemProvider: CompletionItemProvider<CompletionItem> = {
+  provideCompletionItems: (doc, pos) => {
+    const line = doc.lineAt(pos).text;
+    let prefix = line.slice(0, pos.character);
+    if (!prefix.includes('@')) {
+      return [];
+    }
+
+    const txt = doc.getText();
+    const jsDocCommentMatcher = /\/\*\*[^*](?:\r|\n|.)*?\*\//g;
+    let match = jsDocCommentMatcher.exec(txt);
+    let offset = doc.offsetAt(pos);
+    while (match) {
```

Comment:
not sure it is actually more performant than a `matchAll` that would easier to read